src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -38,61 +38,64 @@ import { LoadingListener } from '../../utils/loading/loading-listener';
     trigger('validated', [
       transition(':enter', [
         animate(
-          '300ms',
+          '600ms',
           keyframes([
-            style({ transform: 'scale(0,0)' }),
-            style({ opacity: 1 }),
-            style({ transform: 'scale(1.2,1.2)' }),
-            style({ transform: 'scale(.9,.9)' }),
-            style({ transform: 'scale(1,1)' }),
+            style({ transform: 'scale(0,0)', offset: 0 }),
+            style({ opacity: 1, offset: 0.2 }),
+            style({ transform: 'scale(1.2,1.2)', offset: 0.4 }),
+            style({ transform: 'scale(.9,.9)', offset: 0.6 }),
+            style({ transform: 'scale(1,1)', offset: 1 }),
           ])
         ),
       ]),
       transition(':leave', [style({ opacity: 1 }), animate('100ms ease-out', style({ opacity: 0 }))]),
     ]),
   ],
+  host: { '[attr.disabled]': "disabled? '' : null" },
 })
 export class ClrLoadingButton implements LoadingListener {
   public buttonState = ClrLoadingState;
   public state: ClrLoadingState = ClrLoadingState.DEFAULT;
 
+  @ContentChild(ClrLoading) public loadingDirective: ClrLoading;
+
+  @Input('disabled') public disabled: boolean;
+
   @Output('clrLoadingChange')
   public clrLoadingChange: EventEmitter<ClrLoadingState> = new EventEmitter<ClrLoadingState>(false);
 
-  constructor(private el: ElementRef, private renderer: Renderer2) {}
+  constructor(public el: ElementRef, private renderer: Renderer2) {}
 
   loadingStateChange(state: ClrLoadingState): void {
     this.state = state;
 
     switch (state) {
       case ClrLoadingState.DEFAULT:
         this.renderer.removeStyle(this.el.nativeElement, 'width');
-        this.renderer.removeAttribute(this.el.nativeElement, 'disabled');
+        if (!this.disabled) {
+          this.renderer.removeAttribute(this.el.nativeElement, 'disabled');
+        }
         break;
       case ClrLoadingState.LOADING:
         this.setExplicitButtonWidth();
         this.renderer.setAttribute(this.el.nativeElement, 'disabled', '');
         break;
       case ClrLoadingState.SUCCESS:
         this.setExplicitButtonWidth();
-        setTimeout(() => {
-          this.loadingStateChange(ClrLoadingState.DEFAULT);
-        }, 1000);
         break;
       case ClrLoadingState.ERROR:
         this.loadingStateChange(ClrLoadingState.DEFAULT);
         break;
       default:
         break;
     }
-
     this.clrLoadingChange.emit(state);
   }
 
   private setExplicitButtonWidth() {
-    if (getComputedStyle) {
-      const width = getComputedStyle(this.el.nativeElement).getPropertyValue('width');
-      this.renderer.setStyle(this.el.nativeElement, 'width', width);
+    if (this.el.nativeElement.getBoundingClientRect) {
```

Comment:
Would this work when the server side rendering script runs?

Reply:
Also I believe the `dom-adapter.ts` already has this, no?

Reply:
Good point. I think `dom-adapter.ts` returns the integer value of the width though, which is going to be less precise, and I'm wondering if that's going to shrink the button a tiny bit and result in a tiny shrink/expand of the button. I also am not sure if we should use it outside of datagrid (I think it could be, just don't see an example of us using it so far outside of it)?

As for the server side rendering, I *think* `elementRef` works fine but will defer to @gnomeontherun to confirm if the above code is safe for SSR.

Reply:
This would fail if someone set the default state to `SUCCESS`, which seems unlikely but possible. At the least, you would need to check if `nativeElement` is set in this conditional for SSR.

Reply:
You mean if they started off the button state as `SUCCESS`? That is true, but not something we can accommodate as there would be no way to measure the width with the text without having rendered that first.

Will update to check if `this.el.nativeElement` exists for SSR.